operator/internal/handlers/lokistack_create_or_update.go

```diff
@@ -250,6 +250,20 @@ func CreateOrUpdateLokiStack(
 
 	opts.TLSProfileSpec = spec
 
+	if stack.Spec.Tenants != nil && opts.Stack.Tenants.Mode == lokiv1.OpenshiftLogging {
+		var svc corev1.Service
+		key := client.ObjectKey{Name: manifests.MonitoringSVCOperated, Namespace: manifests.MonitoringNS}
+
+		err = k.Get(ctx, key, &svc)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return kverrors.Wrap(err, "failed to lookup alertmanager service", "name", key)
+		}
+
+		if err == nil {
+			opts.Ruler.OCPAlertManagerEnabled = true
+		}
+	}
```

Comment:
I suggest to push this code block in a separate package internal to handlers package. This handler is already too big and needs some uncluttering, let's start with new additions like this one.